### PR TITLE
Interpret wildcard without let as sugar for let wildcard

### PR DESCRIPTION
**Pull request by peter-evans**
Closes https://github.com/hylo-lang/hylo/issues/977

- Returns `.let` as the introducer from `parseBindingIntroducer` when the current token is `_`

### CODE REVIEW

File: Sources/FrontEnd/Parse/Parser.swift [hunk 1]
```diff
@@ -2624,6 +2624,13 @@ public enum Parser {
   ) throws -> SourceRepresentable<BindingPattern.Introducer>? {
     guard let head = state.peek() else { return nil }
 
+    // Interpret `_ = rhs` as a sugar for `let _ = rhs`
```
Comment (kyouko-taiga) [inline Sources/FrontEnd/Parse/Parser.swift:2627]: That's a clever change but perhaps we should add a note somewhere to say that applying `parseBindingIntroducer` should not be called *before* the parser checked whether the next token was `_` if there's another rule that could apply. That's the case in `parsePattern`, for example.

```swift
// `_` is always a wildcard pattern.
private static func parsePattern(in state: inout ParserState) throws -> AnyPatternID? {
   if let u = state.take(.under) {
    return AnyPatternID(state.insert(WildcardPattern(site: u.site)))
  }

  // Attempt to parse a binding pattern.
  if let p = try parseBindingPattern(in: &state) {
    ...
```

Observe that if we move `parseBindingPattern` at the top of the function, the parser will commit to recognize a binding pattern and fail on inputs like `(first: _, second: let s)`.

**Review by kyouko-taiga — CHANGES_REQUESTED** [review]
Very nice, I didn't expect it would be that simple!

I just think you should add a note in the docs of `parseBindingIntroducer` to warn about the fact that this parser now consumes `_` if it's the next token of the stream.

File: Sources/FrontEnd/Parse/Parser.swift (hunk 1 above)
Comment (peter-evans) [reply Sources/FrontEnd/Parse/Parser.swift:2630]: Added a note https://github.com/hylo-lang/hylo/pull/1348/commits/989caa0e37b00b8ca80a6ea2983a8430d41ea34e

**Review by kyouko-taiga — APPROVED** [review]
LGTM

**Pull request merged**